internal/auth/auth.go

```diff
@@ -249,3 +254,39 @@ func parseTenant(accessToken string) (tenant, domain string, err error) {
 	}
 	return "", "", fmt.Errorf("audience not found for %s", audiencePath)
 }
+
+// ClientCredentials encapsulates all data to facilitate access token creation with client credentials (client ID and client secret)
+type ClientCredentials struct {
```

Comment:
Would've used the `clientcredentials.Config` type but I wanted to encapsulate the token URL formulation logic within this function.